alchemy/src/cloudflare/bundle/plugin-node-compat.ts

```diff
@@ -0,0 +1,113 @@
+import type { Plugin } from "esbuild";
+import type { NodeJSCompatMode } from "miniflare";
+import { relative } from "node:path";
+import chalk from "picocolors";
+import { dedent } from "../../util/dedent.ts";
+import { logger } from "../../util/logger.ts";
+
+/**
+ * An esbuild plugin that will:
+ * - mark any `node:...` imports as external
+ * - warn if there are node imports (if not in v1 mode)
+ *
+ * Applies to: null, als, legacy and v1 modes.
+ */
+export const esbuildPluginNodeCompatImports = (
+  mode: NodeJSCompatMode,
+): Plugin => ({
+  name: "nodejs_compat-imports",
+  setup(pluginBuild) {
+    // Infinite loop detection
+    const seen = new Set<string>();
+
+    // Prevent multiple warnings per package
+    const warnedPackages = new Map<string, string[]>();
+
+    pluginBuild.onStart(() => {
+      seen.clear();
+      warnedPackages.clear();
+    });
+    pluginBuild.onResolve(
+      { filter: /node:.*/ },
```

Comment:
what about `"fs/promises"` instead of `node:fs/promises`

Reply:
Ah, so the reason this wasn't needed in Wrangler is because esbuild catches it and treats it as a `node:` import. But then it doesn't trigger our warning. Fixed.